NG CD RN jan 31 draft (#548)
harness/developer-hub: release-notes/continuous-delivery.md

Additionally, the release notes below are only for NextGen SaaS. FirstGen SaaS release notes are available [here](/docs/first-gen/firstgen-release-notes/harness-saa-s-release-notes) and Self-Managed Enterprise Edition release notes are available [here](/release-notes/self-managed-enterprise-edition).
:::

> ## February 1, 2023, version 78300

Reviewer: Remember to update the date here and in "date:" on line 3. The "date:" on line 3 must be updated with every new release date because it is used to update the RSS.

Author: understood. thanks

  The file store was not showing up at the projects level. This has been fixed and now you can use the file store in your projects.

  ![file store](static/e6e71ed5ce6113726c8385b19408341220d153944a0683dbc76614e7a6aeed9d.png)
- Runtime **Tag** setting discrepancy when switching between YAML and Visual views.	(CDS-50033)
  
  When a user set the **Tag** setting in the Visual view and then switched to the YAML view, made an edit, and switched back, the value was not preserved. Now users can select tags or enter values from YAML without any discrepancies.
- The PagerDuty notifications are not showing start and end dates for pipeline/stage execution. (CDS-49852)

  The PagerDuty template was using the wrong placeholder for [PagerDuty notifications](https://developer.harness.io/docs/continuous-delivery/cd-advanced/cd-notifications/notify-users-of-pipeline-events/#option-pagerduty-notifications). The template is now modified to use the correct placeholder name.

> - Deployment freeze recurrence time should be greater than start time. (CDS-49840)

Reviewer 2: ```suggestion
- The deployment freeze recurrence time should be greater than the start time. (CDS-49840)
```

  
  Harness was letting users set a recurrence time in the past. We have added a check to prevent users from creating a freeze window with a recurrence time that is before than the current time.
- An empty trigger name is not disabling the **Continue** button. (CDS-49631)
  
  When the trigger **Name** was empty a validation error appeared but the **Continue** button could still be clicked.
  
  Now, when the trigger **Name** is empty a validation error appears and the **Continue** button is disabled.

> - **Infrastructure** setting retained in the UI when environment is marked as **Runtime Input**. (CDS-49236)

Reviewer 2: ```suggestion
- The **Infrastructure** setting is retained in the UI when the environment is marked as **Runtime Input**. (CDS-49236)
```

  
  Now the **Runtime Input** setting is maintained and the **Infrastructure** setting is a runtime input.
- Different artifacts with the same build Id are not considered in services listed in the **Environments** dashboard. (CDS-49189)
  
  Earlier, only the build Id was used to group instances in the **Environments** dashboard. When a user deployed a service with the same build Id but different artifact paths, either of the artifact paths would be present in the hover over the build Id in **Artifacts**.

>   Now we group services on their display name (artifact path + build Id) so that there are different entries for different artifacts even if they have the same build Id.

Reviewer 2: ```suggestion
  Now we group services based on their display name (artifact path + build Id) so that there are different entries for different artifacts even if they have the same build Id.
```

  This is now fixed and you are returned to the file.
- The Kubernetes namespace in **Infrastructure** cannot contain capital letters, but they were allowed in Harness YAML. (CDS-48514)
  
  Added schema validation for Kubernetes infrastructure **Namespace** setting to match Kubernetes requirements (`[a-z0-9]([-a-z0-9]*[a-z0-9])`).
- Deployment freeze was missing a check on timezone and accepting the wrong value in YAML. (CDS-48311)
  
  Now there is a check to validate if the timezone entered in the freeze window is correct.
- After fixing runtime input settings in the pipeline **Run** form YAML, users cannot run the pipeline. (CDS-48009)
  
  The pipeline **Run** form accepts Visual and YAML entry. When you entered runtime input values in the YAML the **Run** button was disabled.
  
  Now you can submit runtime input values as Visual or YAML entries and run the pipeline.
- Connector settings do not display in the **Run** pipeline form. (CDS-46632)
  
  When users selected an environment and infrastructure, cleared the selected environment, and then reselected the same environment and infrastructure, connector settings did not display in the **Run** pipeline form.
    

>   Now, the connector setting appear when reselecting the environment and infrastructure.

Reviewer 2: ```suggestion
  Now, the connector settings appear when you reselect the environment and infrastructure.
```
